OpenTelemetry Javaagent failed to start

The opentelemetry java agent failed to start with the following exception:

```
[otel.javaagent 2022-12-13 10:27:53:164 +0000] [main] INFO io.opentelemetry.javaagent.tooling.VersionLogger - opentelemetry-javaagent - version: 1.20.2
[otel.javaagent 2022-12-13 10:27:55:000 +0000] [main] INFO io.opentelemetry.javaagent.shaded.instrumentation.jmx.engine.JmxMetricInsight - Empty JMX configuration, no metrics will be collected for InstrumentationScope io.opentelemetry.jmx
OpenTelemetry Javaagent failed to start
java.lang.NoClassDefFoundError: com/sun/management/GarbageCollectionNotificationInfo
	at io.opentelemetry.javaagent.shaded.instrumentation.runtimemetrics.GarbageCollector.registerObservers(GarbageCollector.java:48)
	at io.opentelemetry.javaagent.shaded.instrumentation.javaagent.runtimemetrics.RuntimeMetricsInstaller.afterAgent(RuntimeMetricsInstaller.java:41)
	at io.opentelemetry.javaagent.tooling.AgentInstaller.runAfterAgentListeners(AgentInstaller.java:263)
	at io.opentelemetry.javaagent.tooling.AgentInstaller.installBytebuddyAgent(AgentInstaller.java:181)
	at io.opentelemetry.javaagent.tooling.AgentInstaller.installBytebuddyAgent(AgentInstaller.java:94)
	at io.opentelemetry.javaagent.tooling.AgentStarterImpl.internalStart(AgentStarterImpl.java:86)
	at io.opentelemetry.javaagent.tooling.AgentStarterImpl.start(AgentStarterImpl.java:67)
	at io.opentelemetry.javaagent.bootstrap.AgentInitializer.initialize(AgentInitializer.java:35)
	at io.opentelemetry.javaagent.OpenTelemetryAgent.startAgent(OpenTelemetryAgent.java:57)
	at io.opentelemetry.javaagent.OpenTelemetryAgent.premain(OpenTelemetryAgent.java:45)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(Unknown Source)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(Unknown Source)
Caused by: java.lang.ClassNotFoundException: com.sun.management.GarbageCollectionNotificationInfo
	at java.base/java.net.URLClassLoader.findClass(Unknown Source)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	at io.opentelemetry.javaagent.bootstrap.AgentClassLoader.loadClass(AgentClassLoader.java:148)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
```

I use this version from maven: io.opentelemetry.javaagent:opentelemetry-javaagent:1.20.2

`java -javaagent:/opt/opentelemetry/opentelemetry-javaagent-1.20.2.jar -jar myapp.jar`

**Environment**
Compiler: openjdk 17
OS: Linux
Build Docker image :  maven:3-openjdk-17
Runtime Docker image : redhat/ubi9/ubi-minimal with a custom JRE built with jlink.

```RUN jlink --add-modules 
java.base,java.desktop,java.instrument,java.logging,java.management,java.naming,java.prefs,java.rmi,java.scripting,java.security.sasl,java.sql,java.sql.rowset,java.transaction.xa,java.xml,jdk.attach,jdk.httpserver,jdk.jdi,jdk.security.jgss,jdk.unsupported,jdk.crypto.ec \
         --strip-debug \
         --no-man-pages \
         --no-header-files \
         --compress=2 \
         --output /work/jre
```

--- a/instrumentation/runtime-metrics/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/GarbageCollector.java
+++ b/instrumentation/runtime-metrics/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/GarbageCollector.java
@@ -122,7 +122,9 @@ public final class GarbageCollector {
 
   private static boolean isNotificationClassPresent() {
     try {
-      Class.forName("com.sun.management.GarbageCollectionNotificationInfo", false,
+      Class.forName(
+          "com.sun.management.GarbageCollectionNotificationInfo",
+          false,
           GarbageCollectorMXBean.class.getClassLoader());
       return true;
     } catch (ClassNotFoundException e) {

--- a/instrumentation/runtime-metrics/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/GarbageCollector.java
+++ b/instrumentation/runtime-metrics/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/GarbageCollector.java
@@ -15,6 +15,7 @@ import java.lang.management.GarbageCollectorMXBean;
 import java.lang.management.ManagementFactory;
 import java.util.List;
 import java.util.function.Function;
+import java.util.logging.Logger;
 import javax.management.Notification;
 import javax.management.NotificationEmitter;
 import javax.management.NotificationFilter;
@@ -32,6 +33,8 @@ import javax.management.openmbean.CompositeData;
  */
 public final class GarbageCollector {
 
+  private static final Logger logger = Logger.getLogger(GarbageCollector.class.getName());
+
   private static final AttributeKey<String> GC_KEY = AttributeKey.stringKey("gc");
   private static final AttributeKey<String> ACTION_KEY = AttributeKey.stringKey("action");
 
@@ -43,6 +46,13 @@ public final class GarbageCollector {
 
   /** Register observers for java runtime memory metrics. */
   public static void registerObservers(OpenTelemetry openTelemetry) {
+    if (!isNotificationClassPresent()) {
+      logger.fine(
+          "The com.sun.management.GarbageCollectionNotificationInfo class is not available;"
+              + " GC metrics will not be reported.");
+      return;
+    }
+
     registerObservers(
         openTelemetry,
         ManagementFactory.getGarbageCollectorMXBeans(),
@@ -108,6 +118,16 @@ public final class GarbageCollector {
   private static GarbageCollectionNotificationInfo extractNotificationInfo(
       Notification notification) {
     return GarbageCollectionNotificationInfo.from((CompositeData) notification.getUserData());
+  }
+
+  private static boolean isNotificationClassPresent() {
+    try {
+      Class.forName("com.sun.management.GarbageCollectionNotificationInfo", false,
+          GarbageCollectorMXBean.class.getClassLoader());
+      return true;
+    } catch (ClassNotFoundException e) {
+      return false;
+    }
   }
 
   private GarbageCollector() {}

